evaluator.go

```diff
@@ -73,12 +78,24 @@ func (s *State) Clusters() map[string]Cluster {
 	return s.clusters
 }
 
+// enable clusterQueryResultCaching
```

Comment:
considering a scenario of 5000 clusters and 6000 nodes, and node length=50 bytes, cluster name length = 50bytes,
MemoryRequiredForCaching = `no.Nodes*sizeOfnode (memoryForKeys) + NumberOfNodes*NumberOfClustersNodeisPartOf*SizeOfCluster (memoryForValue)`

noOfClustersNodeIsPartOf | MemoryRequiredInMb
--------------------------- | ----------------------
1|0.6
10|3.3
50|15.3
100|30.3
1000|300.3
5000|1500.3

But common scenarios, a node is part of 10-50 clusters, so increase in memory would be around ~3-15Mb + memoryOverheadofManaging concurrentMaps another 15Mb?